website_sale_hide_price/views/website_sale_template.xml

```diff
@@ -1,5 +1,15 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <odoo>
+    <template id="products" inherit_id="website_sale.products">
+        <xpath
+            expr="//div[hasclass('o_wsale_products_grid_table_wrapper')]"
+            position="attributes"
+        >
+            <attribute
+                name="t-cache"
+            >pricelist,products,website,website.website_show_price</attribute>
```

Comment:
Why is this needed? A comment should help to clarify for avoiding to remove it in future revisions.

Reply:
This is needed to reload cache in case website_show_price changes (this is a compute which depends on the user being connected whether he is allowed to see prices are not)
Without this, the product list is kept in cache (with or without prices), so you get the exact same list when connecting with different users which do not have the same rights to see price (you get the prices (until next Odoo restart) if the first user to request page after restart of Odoo is allowed to see prices, no price if the first user to request page is not allowed)
This fixes the issue #818 
I added short explanation and link to issue in XML file as suggested